### PR TITLE
fix: typed audiences should take priority.

### PR DESCRIPTION
**Pull request by msohailhussain**
## Summary
- Typed audiences should be taken priority, if any audienceId is same in audiences and typedAudiences, typedAudiences should be considered.

## Test plan
- Updated unit test and FSC should be passed with new testcases.

### CODE REVIEW

File: OptimizelySDK/OptlyConfig/OptimizelyConfigService.cs [hunk 1]
```diff
@@ -74,11 +74,14 @@ private OptimizelyEvent[] GetEvents(ProjectConfig projectConfig)
 
         private OptimizelyAudience[] GetAudiences(ProjectConfig projectConfig)
         {
-            var filteredAudiencesArr = Array.FindAll(projectConfig.Audiences, aud => !aud.Id.Equals("$opt_dummy_audience"));
-            var optimizelyAudience = filteredAudiencesArr.Select(aud => new OptimizelyAudience(aud.Id, aud.Name, aud.Conditions));
             var typedAudiences = projectConfig.TypedAudiences?.Select(aud => new OptimizelyAudience(aud.Id,
                 aud.Name,
                 JsonConvert.SerializeObject(aud.Conditions)));
+            var typedAudienceIds = typedAudiences.Select(ta => ta.Id).ToList();
+            var filteredAudiencesArr = Array.FindAll(projectConfig.Audiences, aud => !aud.Id.Equals("$opt_dummy_audience")
```
Comment (The-inside-man) [inline OptimizelySDK/OptlyConfig/OptimizelyConfigService.cs:81]: Does this keep precedence for typedAudiences in filteredAudiencesArr?

**Comment by The-inside-man** [comment]
LGTM - Just noting the FSC failure due to the FSC fix not merged yet. Noted that is the only reason it is failing.

**Review by The-inside-man — APPROVED** [review]

**Review by jaeopt — APPROVED** [review]
LGTM

**Comment by msohailhussain** [comment]
PR is passing.
https://travis-ci.com/github/optimizely/fullstack-sdk-compatibility-suite/builds/234629631

**Pull request merged**